entity-framework/core/modeling/query-types.md

```diff
@@ -1,53 +1,49 @@
-- Are addressed on the _ModelBuilder_ using the `Query` method rather than the `Entity` method.
-- Are mapped on the _DbContext_ through properties of type `DbQuery<T>` rather than `DbSet<T>`
-- Are mapped to database objects using the `ToView` method, rather than `ToTable`.
-- May be mapped to a _defining query_ - A defining query is a secondary query declared in the model that acts a data source for a query type.
+  - Entities cannot contain navigation properties to keyless entity types.
+- Need to be configured with `.HasNoKey()` method call.
+- May be mapped to a _defining query_. A defining query is a query declared in the model that acts as a data source for a keyless entity type.
 
 ## Usage scenarios
 
-Some of the main usage scenarios for query types are:
+Some of the main usage scenarios for keyless entity types are:
 
-- Serving as the return type for ad hoc `FromSql()` queries.
-- Mapping to database views.
+- Serving as the return type for `FromSql()` queries.
+- Mapping to database views that do not contain a primary key.
 - Mapping to tables that do not have a primary key defined.
 - Mapping to queries defined in the model.
 
 ## Mapping to database objects
 
-Mapping a query type to a database object is achieved using the `ToView` fluent API. From the perspective of EF Core, the database object specified in this method is a _view_, meaning that it is treated as a read-only query source and cannot be the target of update, insert or delete operations. However, this does not mean that the database object is actually required to be a database view - It can alternatively be a database table that will be treated as read-only. Conversely, for entity types, EF Core assumes that a database object specified in the `ToTable` method can be treated as a _table_, meaning that it can be used as a query source but also targeted by update, delete and insert operations. In fact, you can specify the name of a database view in `ToTable` and everything should work fine as long as the view is configured to be updatable on the database.
+Mapping a keyless entity type to a database object is achieved using the `ToTable` or `ToView` fluent API. From the perspective of EF Core, the database object specified in this method is a _view_, meaning that it is treated as a read-only query source and cannot be the target of update, insert or delete operations. However, this does not mean that the database object is actually required to be a database view. It can alternatively be a database table that will be treated as read-only. Conversely, for normal entity types, EF Core assumes that a database object specified in the `ToTable` method can be treated as a _table_, meaning that it can be used as a query source but also targeted by update, delete and insert operations. In fact, you can specify the name of a database view in `ToTable` and everything should work fine as long as the view is configured to be updatable on the database.
```

Comment:
Should also specify differences in migrations if any.

Reply:
Also, this whole paragraph applies to regular entity types as well, so should it be replicated somewhere else in the docs? Or maybe moved into its own central place (in the mapping to tables section that @roji is working on in https://github.com/aspnet/EntityFramework.Docs/pull/1717) and then linked from here?

Reply:
I'd say let's merge this as-is for now (we're mainly adjusting query types ->  keyless in this PR) and revisit refactoring this later.